buildpack/build.go

```diff
@@ -81,13 +82,57 @@ func (b *Descriptor) Build(bpPlan Plan, config BuildConfig) (BuildResult, error)
 		return BuildResult{}, err
 	}
 
+	if err := b.checkTypesFormat(bpLayersDir, config.Out); err != nil {
+		return BuildResult{}, err
+	}
+
 	if err := b.setupEnv(config.Env, bpLayersDir); err != nil {
 		return BuildResult{}, err
 	}
 
 	return b.readOutputFiles(bpLayersDir, bpPlanPath, bpPlan, config.Out)
 }
 
+func renameLayerDirIfNeeded(layerMetadataFile layertypes.LayerMetadataFile, layerDir string) error {
+	// rename <layers>/<layer> to <layers>/<layer>.ignore if buildpack API >= 0.6 and all of the types flags are set to false
+	if !layerMetadataFile.Launch && !layerMetadataFile.Cache && !layerMetadataFile.Build {
+		if err := os.Rename(layerDir, layerDir+".ignore"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *Descriptor) checkTypesFormat(layersDir string, out io.Writer) error {
```

Comment:
It seems this function is doing a bit more than just checking the types, perhaps the name could be more descriptive?

Reply:
I agree but I'm not sure what is the right name.
The function:
- Checks the types format and error/warn if the format is wrong
- Rename the layer directory to `.ignore` if the buildpack API >= 0.6 and all the flags are set to false.

What do you think about `doTypesManupulations`?
Any better ideas?

Reply:
Hmm... `processLayerTypes`? IDK!

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/d6aa6e72ae980771d30efd95383086f0e1995f67.